components/kyma-environment-broker/internal/cls/provisioner_test.go

```diff
@@ -0,0 +1,178 @@
+package cls
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/cls/automock"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/logger"
+	smautomock "github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/servicemanager/automock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProvisionCreatesNewInstanceIfNoneFoundInDB(t *testing.T) {
+	const (
+		fakeGlobalAccountID = "fake-global-account-id"
+		fakeSKRInstanceID   = "fake-skr-instance-id"
+		fakeBrokerID        = "fake-broker-id"
+		fakeServiceID       = "fake-service-id"
+		fakePlanID          = "fake-plan-id"
+	)
+
+	storageMock := &automock.InstanceStorage{}
```

Comment:
use in memory implementation instead of mock (we don't need mock for the storage)

Reply:
one advantage of using mocks in this case that I see is that I can test e.g. DB connection problems (when the storage returns `dberrors`)

Reply:
for instance, this test case: https://github.com/kyma-project/control-plane/blob/e691b02a17f411333bfd2dcad0579f7cc5abe7fb/components/kyma-environment-broker/internal/cls/provisioner_test.go#L79 tests that an instance is not created if it cannot be saved to the DB first

Reply:
for errors using a mock looks fine, for happy path - you can use in-memory